clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml

```diff
@@ -124,6 +124,73 @@
       In: ${arg}.cu
       Out: ${arg}.dp.cpp
 
+- Rule: rule_cuda_compile_with_opts
+  Kind: CMakeRule
+  Priority: Fallback
+  MatchMode: Partial
+  CmakeSyntax: rule_cuda_compile_01
+  In: cuda_compile(${libname} ${srcs} STATIC OPTIONS ${opts})
+  Out: dpct_compile(${libname} ${srcs})
+  Subrules:
+    srcs:
+      MatchMode: Full
+      In: ${arg}.cu
+      Out: ${arg}.dp.cpp
+
+- Rule: rule_cuda_compile_with_opts
+  Kind: CMakeRule
+  Priority: Fallback
+  MatchMode: Partial
+  CmakeSyntax: rule_cuda_compile_02
+  In: cuda_compile(${libname} ${srcs} SHARED OPTIONS ${opts})
+  Out: dpct_compile(${libname} ${srcs})
+  Subrules:
+    srcs:
+      MatchMode: Full
+      In: ${arg}.cu
+      Out: ${arg}.dp.cpp
+
+
+- Rule: rule_cuda_compile_with_lib_type
+  Kind: CMakeRule
+  Priority: Fallback
+  MatchMode: Partial
+  CmakeSyntax: rule_cuda_compile_03
+  In: cuda_compile(${libname} ${srcs} STATIC)
+  Out: dpct_compile(${libname} ${srcs})
+  Subrules:
+    srcs:
+      MatchMode: Full
+      In: ${arg}.cu
+      Out: ${arg}.dp.cpp
+
+- Rule: rule_cuda_compile_with_lib_type
+  Kind: CMakeRule
+  Priority: Fallback
+  MatchMode: Partial
+  CmakeSyntax: rule_cuda_compile_04
+  In: cuda_compile(${libname} ${srcs} SHARED)
+  Out: dpct_compile(${libname} ${srcs})
+  Subrules:
+    srcs:
+      MatchMode: Full
+      In: ${arg}.cu
+      Out: ${arg}.dp.cpp
+
+
+- Rule: rule_cuda_compile_with_no_opts
+  Kind: CMakeRule
+  Priority: Fallback
+  MatchMode: Partial
+  CmakeSyntax: rule_cuda_compile_05
+  In: cuda_compile(${libname} ${srcs})
+  Out: dpct_compile(${libname} ${srcs})
```

Comment:
I'd prefer to dpct_helper_sycl_compile.  let's update the name style in a follow up PR.

Reply:
Sounds correct! Modified.